sources/packages/backend/apps/api/src/services/student-assessment/student-assessment.service.ts

```diff
@@ -236,4 +249,94 @@ export class StudentAssessmentService extends RecordDataModelService<StudentAsse
       .addOrderBy("assessment.submittedDate", "DESC")
       .getMany();
   }
+
+  /**
+   * Creates a manual assessment for the application.
+   * @param applicationId application id.
+   * @param note note.
+   * @param userId user id who triggered the manual reassessment.
+   * @returns the id of the assessment created.
+   */
+  async manualReassessment(
+    applicationId: number,
+    note: string,
+    userId: number,
+  ): Promise<StudentAssessment> {
+    return this.dataSource.transaction(async (transactionalEntityManager) => {
+      const application =
+        await this.applicationService.getApplicationAssessmentStatusDetails(
+          applicationId,
+          { entityManager: transactionalEntityManager },
+        );
+
+      if (!application) {
+        throw new CustomNamedError(
+          `Application not found.`,
+          APPLICATION_NOT_FOUND,
+        );
+      }
+      if (application.isArchived) {
+        throw new CustomNamedError(
+          `Application cannot have manual reassessment after being archived.`,
+          INVALID_OPERATION_IN_THE_CURRENT_STATUS,
+        );
+      }
+      if (
+        [
+          ApplicationStatus.Cancelled,
+          ApplicationStatus.Overwritten,
+          ApplicationStatus.Draft,
+        ].includes(application.applicationStatus)
+      ) {
+        throw new CustomNamedError(
+          `Application cannot have manual reassessment in any of the statuses: ${ApplicationStatus.Cancelled}, ${ApplicationStatus.Overwritten} or ${ApplicationStatus.Draft}.`,
+          INVALID_OPERATION_IN_THE_CURRENT_STATUS,
+        );
+      }
+      const [originalAssessment] = application.studentAssessments;
+      if (
+        originalAssessment.studentAssessmentStatus !==
+        StudentAssessmentStatus.Completed
+      ) {
+        throw new CustomNamedError(
+          `Application original assessment expected to be '${StudentAssessmentStatus.Completed}' to allow manual reassessment.`,
+          INVALID_OPERATION_IN_THE_CURRENT_STATUS,
+        );
+      }
+
+      await this.noteSharedService.createStudentNote(
+        application.student.id,
+        NoteType.Application,
+        note,
+        userId,
+        transactionalEntityManager,
+      );
+
+      const auditUser = { id: userId } as User;
+      const now = new Date();
+      const oldCurrentAssessment = application.currentAssessment;
+      const applicationToBeSaved = {
+        id: application.id,
+        modifier: auditUser,
+        updatedAt: now,
+      } as Application;
+      applicationToBeSaved.currentAssessment = {
+        application: applicationToBeSaved,
+        offering: { id: oldCurrentAssessment.offering.id },
+        studentAppeal: oldCurrentAssessment?.studentAppeal?.id
```

Comment:
Quick question, can it be just 
```ts
studentAppeal: oldCurrentAssessment.studentAppeal
```

Reply:
it does need to check it because studentAppeal is optional. Will use the nullish coalescing operator then.

Reply:
I checked it and it doesn't work with the nullish coalescing operator because I need to set the entire studentAppeal in case `oldCurrentAssessment?.studentAppeal?.id` is defined.